contracts/DCNT721A.sol

```diff
@@ -49,12 +56,25 @@ contract DCNT721A is ERC721A, Initializable, Ownable, Splits {
   /// @param tokenId_ of token minted
   event Minted(address sender, uint256 tokenId_);
 
+  /// ========== Modifier =============
+  modifier verifyTokenGate(bool isPresale) {
+    if (tokenGateConfig.tokenAddress != address(0)
+      && (tokenGateConfig.saleType == SaleType.ALL ||
+          isPresale && tokenGateConfig.saleType == SaleType.PRESALE) ||
+          !isPresale && tokenGateConfig.saleType == SaleType.PRIMARY) {
+            require(ITokenWithBalance(tokenGateConfig.tokenAddress).balanceOf(msg.sender) >= tokenGateConfig.minBalance, 'do not own required token');
```

Comment:
>= / <= costs more gas than > < recommend updating for gas optimization.

https://nftchance.medium.com/the-gas-efficient-way-of-building-and-launching-an-erc721-nft-project-for-2022-b3b1dac5f2e1